# SC_ST_CTB_figures_scripts
A Spatiotemporal Single-Cell Atlas of Cutaneous Tuberculosis Infection and Treatment Response

## Repository Contents
- **scripts** contains scripts used for running differential gene expression analysis, functional enrichment analysis, plotting UMAP plots and marker gene bubble plots, and more.
- This repository contains code modules for advanced analysis of single-cell and spatial transcriptomic data, focusing on cell type deconvolution, cellular communication, and spatial tissue organization. The directory structure and brief descriptions are as follows:
- deconvolution_RCTD/​ - Contains scripts for cell-type deconvolution of spatial transcriptomics data using the RCTD (Robust Cell Type Decomposition) tool.
- decoupleR_score/​ - Includes code for calculating pathway activity scores at the single-cell or spot level using the decoupleRtool.
- pyscenic/​ - Houses the workflow for running pySCENICto infer gene regulatory networks and analyze transcription factor activity.
- single_cell_cellcrosstalk/​ - Dedicated to the analysis of cell-cell communication from single-cell data.
- spatial_distance_calculation/​ - Provides tools for quantifying spatial distances between different cell types or molecular features within a tissue section.
- spatial_niche_analysis/​ - Contains methods for identifying and characterizing spatial cellular niches or microenvironments in tissue.
- spatial_NMF_analysis/​ -  Implements Non-negative Matrix Factorization (NMF) for dimensionality reduction and pattern discovery in spatial transcriptomics data.
## Data Available
- The raw sequencing data from this study have been deposited in the Gene Expression Omnibus (GEO) database under the accession number GSExxxxxx.
